.github/workflows/c-cpp.yml

```diff
@@ -29,24 +29,39 @@ jobs:
         name: macos
         path: 'build/**/**/arqma*'
 
-  build-osx-depends:
-    runs-on: ubuntu-latest
+  build-windows:
+    runs-on: windows-latest
+    env:
+      CCACHE_COMPRESS: 1
+      CCACHE_TEMPDIR: C:\Users\runneradmin\.ccache-temp
+      CCACHE_DIR: C:\Users\runneradmin\.ccache
+      ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+    defaults:
+      run:
+        shell: msys2 {0}
     steps:
     - uses: actions/checkout@v1
       with:
         submodules: recursive
-    - name: install ArQmA dependencies
-      run:  sudo apt -y install cmake ccache imagemagick libcap-dev librsvg2-bin libz-dev libbz2-dev libtiff-tools curl bsdmainutils python3-setuptools
-    - name: add SDK
-      run: git clone -b arqma https://github.com/malbit/MacOSX-SDKs.git contrib/depends/SDKs
+    - uses: actions/cache@v2
+      with:
+        path: C:\Users\runneradmin\.ccache
+        key: ccache-windows-build-${{ github.sha }}
+        restore-keys: ccache-windows-build-
+    - uses: eine/setup-msys2@v2
+      with:
+        update: true
+        install: mingw-w64-x86_64-toolchain make mingw-w64-x86_64-cmake mingw-w64-x86_64-openssl mingw-w64-x86_64-libsodium mingw-w64-x86_64-hidapi mingw-w64-x86_64-libusb mingw-w64-x86_64-libtool binutils git automake autoconf patch mingw-w64-x86_64-ccache
+    - name: boost
+      run: pacman --noconfirm -U https://gitlab.com/ArqTras/depends/-/raw/main/mingw-w64-x86_64-boost-1.73.0-3-any.pkg.tar.zst
```

Comment:
can we use official package from msys/mingw official repository??

Reply:
No cus them was not available for some time so I did moved in to mirror. Basically it’s just workflow file  so have nothing to do with user system

Reply:
Link which I did include in suggestion points to mirrors list and checks which mirror is live and downloading from working one then